keanu-project/src/main/java/io/improbable/keanu/vertices/Vertex.java

```diff
@@ -102,17 +102,32 @@ public final boolean isProbabilistic() {
      * @param value the observed value
      */
     public void setValue(T value) {
-        if (!observation.isObserved()) {
-            this.value = value;
+        if (!state.isObserved()) {
+            state = new VertexState<>(value, false);
         }
     }
 
     @Override
     public T getValue() {
-        return hasValue() ? value : lazyEval();
+        return hasValue() ? state.getValue() : lazyEval();
+    }
+
+    @Override
+    public VertexState getState() {
+        return state;
+    }
+
+    @Override
+    public void setState(VariableState newState) {
```

Comment:
Will anyone ever pass in a `VariableState`? Can we remove the need for:

```
public void setState(VertexState<T> newState) {
        state = newState;
    }
```

If you changed the parameter signature of `setState` to: 
```
<U extends VariableState> void setState(U variableState);
```

Reply:
done